clang/lib/Parse/ParseExpr.cpp

```diff
@@ -1066,6 +1066,20 @@ ExprResult Parser::ParseCastExpression(CastParseKind ParseKind,
     break;
   }
 
+  case tok::annot_embed: {
+    // We've met #embed in a context where a single value is expected. Take last
+    // element from #embed data as if it were a comma expression.
+    EmbedAnnotationData *Data =
+        reinterpret_cast<EmbedAnnotationData *>(Tok.getAnnotationValue());
+    SourceLocation StartLoc = ConsumeAnnotationToken();
+    ASTContext &Context = Actions.getASTContext();
+    Res = IntegerLiteral::Create(Context,
+                                 llvm::APInt(CHAR_BIT, Data->BinaryData.back()),
+                                 Context.UnsignedCharTy, StartLoc);
+    if (Data->BinaryData.size() > 1)
+      Diag(StartLoc, diag::warn_unused_comma_left_operand);
+  } break;
```

Comment:
(Weird formatting)

I think this is correct, thanks
(ie, are there cases where  `number, ` in a cast expression would not be part of a comma expression? Probably not)

Reply:
I put break into the scope aligned by curly braces, hope it is less weird.

> (ie, are there cases where number,  in a cast expression would not be part of a comma expression? Probably not)

I'm not sure there are.